test/unit/templateLiterals.spec.ts

```diff
@@ -60,3 +60,29 @@ test.each(["func`noSelfParameter`", "obj.func`noSelfParameter`", "obj[`func`]`no
         `.expectToMatchJsResult();
     }
 );
+
+test.each([
+    ["string", false],
+    ["'string literal type'", false],
+    ["string & unknown", false],
+    ["number", true],
+    ["string | any", true],
+])("template span expect tostring for type (%p) - %p", (type, expectToString) => {
+    util.testFunction`
+        function func(msg: ${type}) {
```

Comment:
Can this be simplified to something like this?

```ts
let msg: ${type} ;
// @ts-ignore
return  \`\${msg}\`;
```

Reply:
Somewhat, TypeScript is a bit too smart for that case

```ts
let msg: string | number = 5;
msg; // type: number
```

But this gets around it

```ts
let msg = 5 as string | number;
msg; // type: string | number
```